component/storage/indexeddb/indexeddb.go

```diff
@@ -79,18 +96,64 @@ func (p *Provider) OpenStore(name string) (storage.Store, error) {
 		return nil, err
 	}
 
-	return &store{name: name, db: p.stores[name]}, nil
+	return p.stores[name], nil
 }
 
-// SetStoreConfig is currently not implemented. It always returns a nil error so that IndexedDB can still be used as
-// a cache provider in FormatProvider for the time being.
+// SetStoreConfig sets the configuration on a store.
```

Comment:
This "indexing" implementation is intended to be a temporary solution while I work on updating aries-framework-go and TrustBloc.

In order to properly update everything in aries-framework-go to use the new storage interface without breaking the aries-js-worker I need the new IndexedDB implementation to support tags+querying.

The problem is that the proper way to do indexing (using IndexedDB's built in indexing support) requires setting the indexes up on store creation... but the awkward thing is that we need to "pre-create" some stores per #1347. So I need to know all the stores Aries uses and their indexes up front. But I don't know what those indexes are until I do the big task of updating aries-framework-go to use the new storage interface... a bit of a chicken and egg problem.

So, for now, the "indexing" done in this PR is a sort of emulated approach that allows me to update aries-framework-go and test it along with our TrustBloc stuff. The "real" indexing approach will also take a lot longer to implement, and it's quite important that I get aries-framework-go updated to the new storage interface earlier rather than later so I can discover issues early on.

I've created a follow up issue to improve this implementation.

Reply:
(this isn't the only way to approach this problem, but I believe it's easier than the alternative which is trying to locally do `replaces` and then sort of working backwards/forwards creating PRs...)